zipline/finance/risk/period.py

```diff
@@ -0,0 +1,200 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logbook
+
+from six import iteritems
+
+import numpy as np
+import pandas as pd
+
+from empyrical import (
+    alpha_beta_aligned,
+    annual_volatility,
+    cum_returns,
+    downside_risk,
+    max_drawdown,
+    sharpe_ratio,
+    sortino_ratio
+)
+
+log = logbook.Logger('Risk Period')
+
+
+# check if a field in rval is nan, and replace it with
+# None.
+def check_entry(key, value):
+    if key != 'period_label':
+        return np.isnan(value) or np.isinf(value)
+    else:
+        return False
+
+
+class RiskMetricsPeriod(object):
+    def __init__(self, start_session, end_session, returns, trading_calendar,
+                 benchmark_returns, algorithm_leverages=None):
+        self._start_session = start_session
+        self._end_session = end_session
+        self.trading_calendar = trading_calendar
+
+        trading_sessions = trading_calendar.sessions_in_range(
+            self._start_session,
+            self._end_session,
+        )
+        self.algorithm_returns = self.mask_returns_to_period(returns,
+                                                             trading_sessions)
+
+        # Benchmark needs to be masked to the same dates as the algo returns
+        self.benchmark_returns = self.mask_returns_to_period(
+            benchmark_returns,
+            self.algorithm_returns.index
+        )
+        self.algorithm_leverages = algorithm_leverages
```

Comment:
I'm hesitant to change things in this pull request, but it would seem that by not calling `mask_returns_to_period` then the `RiskMetricsPeriod` will always return the global max_leverage rather than the period max_leverage. I'm not sure why that is.